client/homescreen/activity-panel/stock/index.js

```diff
@@ -0,0 +1,160 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Component } from '@wordpress/element';
+import { compose } from '@wordpress/compose';
+import { withDispatch, withSelect } from '@wordpress/data';
+import PropTypes from 'prop-types';
+import { EmptyContent, Section } from '@woocommerce/components';
+import { ITEMS_STORE_NAME } from '@woocommerce/data';
+
+/**
+ * Internal dependencies
+ */
+import { ActivityCardPlaceholder } from '../../../header/activity-panel/activity-card';
+import { ProductStockCard } from './card';
+
+const productsQuery = {
+	page: 1,
+	per_page: 5,
+	low_in_stock: true,
+	status: 'publish',
+	_fields: [
+		'attributes',
+		'id',
+		'images',
+		'last_order_date',
+		'low_stock_amount',
+		'name',
+		'parent_id',
+		'stock_quantity',
+		'type',
+	],
+};
+export class StockPanel extends Component {
+	constructor( props ) {
+		super( props );
+
+		this.updateStock = this.updateStock.bind( this );
+	}
+
+	async updateStock( product, quantity ) {
+		const { invalidateResolution, updateProductStock } = this.props;
+
+		const result = await updateProductStock( product, quantity );
+
+		if ( result.success ) {
+			// Request more low stock products.
+			invalidateResolution( 'getItems', [ 'products', productsQuery ] );
```

Comment:
I did notice the same issue I had with the `reviews` that if you update the stock when there is only one item, and then click `undo` it won't update the homescreen (it did correctly revert the stock).

Although I can't reproduce this every-time only on a fresh page load, and it did eventually update after expanding/minimizing the panel.

I had to invalidate the `getItemsTotalCount` in that case to re-trigger the search.
![update-stock](https://user-images.githubusercontent.com/2240960/100151190-34fe5080-2e77-11eb-8067-e552a2288a7b.gif)

Reply:
I've mitigated this somewhat in https://github.com/woocommerce/woocommerce-admin/pull/5729/commits/d48edc2946faafc9b9e792527d2d9715b94e5af0 by defaulting the panel to closed (as per Slack convo). If you close the panel before clicking "undo" in the notice, the panel won't refresh - that's an edge case that I think can be handled in a follow up PR.